### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by pan64** [koalaman/shellcheck#1379]
#### For bugs
- Rule Id SC2094: 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```
#!/bin/bash
function dav () {
echo -e "What text file will I be using?"
read -er item
f=$item
while IFS="^" read -r f k
do
if [[ ! -f "$f" ]]; then
	echo "I don't see $f in this directory."
	echo "Moving to next list item."
	continue
fi
# some code here
done<"$f"
}
```

#### Here's what shellcheck currently says:
```
Line 8:
if [[ ! -f "$f" ]]; then
           ^-- SC2094: Make sure not to read and write the same file in the same pipeline.
 
Line 14:
done<"$f"
     ^-- SC2094: Make sure not to read and write the same file in the same pipeline.
```
#### Here's what I wanted or expected to see:

the real problem is in line read -r f k, where the variable r is set inside the while loop. This is not reported by shellcheck, but would be better. This message is misleading (or probably this is a new issue)
